hash_practice/exercises.py

```diff
@@ -22,8 +49,29 @@ def valid_sudoku(table):
         Each element can either be a ".", or a digit 1-9
         The same digit cannot appear twice or more in the same 
         row, column or 3x3 subgrid
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n**2)

```

Comment:
This is correct in a very technical sense and would be the answer if we accepted an arbitrary grid, but since we assume it's always a 3x3 grid, it will always take the same time, so we can just say it's O(1).